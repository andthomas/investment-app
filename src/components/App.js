import React, { Component } from "react";
import "./App.less";
import { hot } from "react-hot-loader";
import Home from "../pages/Home.js";
import Login from "../pages/Login.js";
import { Route, Switch } from 'react-router-dom';
import { Grommet } from 'grommet';

class App extends Component {
    render() {
        return (
            <Grommet plain>
                <Route
                    render={({ location }) => {
                        const { pathname } = location;
                        return (
                            <Route
                                location={location}
                                render={() => (
                                <div className="App">
                                    <Switch>
                                        <Route path='/login' component={Login} />
                                        <Route exact path='/' component={Home} />
                                    </Switch>
                                </div>
                                )}
                            />
                        );
                    }}
                />
            </Grommet>
        );
    }
}

export default hot(module)(App);