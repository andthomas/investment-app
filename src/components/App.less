html {
  height: 100%;
  padding: 0;
  overflow: auto;
  margin: 0;
  -webkit-overflow-scrolling: touch;
}

body {
  margin: 0px;
  font-family: 'Josefin Sans', sans-serif;
  height: 100vh;
  overflow: auto;
  overscroll-behavior: none;
  background-image: linear-gradient(120deg, #fdfbfb 0%, #ebedee 100%);
  height: 100%;
  padding: 0;
  overflow: auto;
  margin: 0;
  -webkit-overflow-scrolling: touch;
}

.App {
  overflow: hidden;
  background-color: #fff;
  max-width: 375px;
  max-height: 950px;
  position: relative;
  height: 100vh;
  margin: 0 auto;
}

.page {
  height: 100vh;
  padding: calc(15% + 1.5em) 5% 5%;
  overflow-y: auto;
  position: fixed;
  top: 0;
  width: 100%;
  -webkit-overflow-scrolling: touch;
  transition: transform 1s ease-in-out;
}

.page-enter {
  transform: translate(-100%, 0);
}
.page-enter-active {
  transform: translate(0, 0);
}
.page-exit {
  transform: translate(-100%, 0);
}

::-webkit-scrollbar {
  width: 0px;
  background: transparent;
}