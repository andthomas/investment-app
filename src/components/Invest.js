import React, { Component } from "react";
import "./Invest.less";
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

class Invest extends Component {
    constructor(props) {
        super(props);
        this.state = {
            value: 0,
            visibleShares: [],
        }
    }

    onItemClick(share) {
        let newShare = true
        this.props.selectedShares.forEach( (s, index) => {
            if (share === s) {
                this.props.selectedShares.splice(index, 1);
                newShare = false;
            }
        })
        if (newShare) {
            if (this.props.selectedShares.length == 5) return;
            this.props.selectedShares.push(share);
        }
        this.props.onSelectShares(this.props.selectedShares);
    }

    highlightSelectedButtons() {
        const shareButtons = Array.from(document.getElementsByClassName('inner-grid'))
        let that = this;
        shareButtons.forEach((s) => {
            s.style.boxShadow = "5px 5px 5px rgba(0,0,0,0.2)"

            that.props.selectedShares.forEach((ss, i) => {
                if (s.id === ss) {
                    document.getElementById(ss).style.boxShadow = '5px 5px 5px rgba(0,153,121,1)'
                };
            })
        })
    }

    filterShares(e) {
        const results = this.props.shareList.filter((s, i) => {
            if (s.name.toLowerCase().includes(e.toLowerCase())) return s;
        })
        this.setState({visibleShares: results})
    }

    componentDidMount() {
        this.setState({visibleShares: this.props.shareList});
        this.highlightSelectedButtons()
    }
    
    componentDidUpdate() {
        this.highlightSelectedButtons()
    }

    imgUrl(img) {
        return `https://res.cloudinary.com/djq5ic5br/image/upload/c_scale,h_120/${img}`
    }

    render() {
        return (
            <div className="invest-container">
                <div className="grid">
                    <input
                        placeholder="Filter NASDAQ"
                        onChange={event => this.filterShares(event.target.value)}
                    ></input>
                    {
                        this.state.visibleShares.map( (share) => {
                            return (
                                <div 
                                    className="inner-grid" 
                                    id={share.id}
                                    onClick={() => {this.onItemClick(share.id)}}
                                    key={share.id}>
                                    <img src={this.imgUrl(share.icon)} />
                                        <div>
                                            <p><strong>{share.id}</strong></p>
                                            <p>{share.name}</p>
                                        </div>
                                </div>
                                )
                        })
                    }

                </div>
            </div>
        );
    }
}

Invest.propTypes = {
    selectedShares: PropTypes.array,
    shareList: PropTypes.array
}

function mapStateToProps(state) {
    return {
        shareList: state.data.shareList
    };
}

export default connect(
    mapStateToProps
)(Invest);
