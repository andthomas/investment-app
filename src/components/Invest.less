.invest-container {
    margin: 0 auto;
    position: relative;
    width: 90%;
    height: 101%;
    overflow-y: scroll;

    .grid {
        display: grid;
        grid-template-columns: calc(33.3% - 7px) calc(33.3% - 7px) calc(33.3% - 7px);
        margin: 30px 0px;
        grid-column-gap: 10px;
        grid-row-gap: 10px;
        
        .inner-grid {
            padding: 10px 2px;
            color: #275096;
            border-radius: 5px;
            text-align: center;
            background-color: #ffffff;
            cursor: pointer;
            box-shadow: 5px 5px 5px rgba(0,0,0,0.2);
        }
    }
}