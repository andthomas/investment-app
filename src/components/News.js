import React, { Component } from "react";
import "./News.less";
import { hot } from "react-hot-loader";
import PropTypes from 'prop-types';

class News extends Component {
    constructor(props) {
        super(props)
        this.state = {currentNews: []}
    }
    
    componentDidMount() {
        const req = new Request('https://newsapi.org/v2/top-headlines?sources=financial-times&apiKey=87903eb739404351971c2d3106d16e7e')
        fetch(req)
        .then( (response) => {
            const data = response.json();
            return data;
        })
        .then( (data) => {
            document.getElementsByClassName('loading')[0].style.display = 'none';
            this.setState({currentNews: data.articles})
        })
    }
    render() {
        return (
            <div className={`news`}>
                    <h4 className="loading">LOADING NEWS</h4>
                    {this.state.currentNews.map((item, key) =>
                        <div
                            key={key}>
                            <div className={`news-tile`}>
                                <img className={`news-item`} src={item.urlToImage}/>
                                <div className={`news-title`}>{item.title}</div>
                                <div className={`news-subtitle`}>{item.source.name}</div>
                                <div className={`news-description`}>{item.description}</div>
                            </div>
                        </div>
                    )}
            </div>
        );
    }
}

News.propTypes = {
    currentNews: PropTypes.array
};

export default hot(module)(News);
