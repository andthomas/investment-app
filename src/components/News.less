.news {
    color: #fff;
    padding: 0px 20px;
    // padding-top: 40px;
    overflow: scroll;
    max-height: 84vh;
    font-weight: 300;

    h2 {
        font-weight: 500;
        text-align: center;
    }
    .news-tile {
        margin-bottom: 20px;
        position: relative;
        width: 100%;
        background-color: white;
        border-radius: 10px;
        font-family: 'Roboto', sans-serif;
        box-shadow: 5px 5px 5px rgba(0,0,0,0.2);

        .news-item {
            width: 100%;
            position: relative;
            border-radius: 10px 10px 0px 0px;    
        }

        .news-title {
            padding: 10px;
            color: black;
            font-size: 16px;
        }
    }
}