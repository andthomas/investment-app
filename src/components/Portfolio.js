import React, { Component } from "react";
import "./Portfolio.less";
import { hot } from "react-hot-loader";
import { Meter, Chart } from "grommet";

class Portfolio extends Component {
    render() {
        return (
            <div className="portfolio-container">
                
                <div className="meter-container">
                    <Meter
                        className="meter"
                        values={[{
                            value: 60,
                            color: 'white',
                            label: 'sixty',
                            onClick: () => { }
                        }]}
                        thickness='small'
                        type="circle"
                        aria-label="meter"
                        />
                        <div className="meter-label">
                            <span>Portfolio value</span>
                            <p>$6,103</p>
                            <span>61% of $10,000 goal</span>
                        </div>
                </div>

                <div className="chart-container">
                    {/* <Chart
                        type="line"
                        bounds={[[0, 7], [0, 100]]}
                        values={[
                            { value: [7, 100], label: 'one hundred' },
                            { value: [6, 70], label: 'seventy' },
                            { value: [5, 60], label: 'sixty' },
                            { value: [4, 80], label: 'eighty' },
                            { value: [3, 40], label: 'forty' },
                            { value: [2, 0], label: 'zero' },
                            { value: [1, 30], label: 'thirty' },
                            { value: [0, 60], label: 'sixty' },
                        ]}
                        aria-label="chart"
                    /> */}

                </div>
            </div>
        )
    }
}

export default hot(module)(Portfolio);
