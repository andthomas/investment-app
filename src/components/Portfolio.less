.portfolio-container {
    margin: 0 auto;
    position: relative;
    width: 90%;
    height: 100%;
    overflow-y: scroll;

    .meter-container {
        position: relative;
        width: 50%;
        margin: 0 auto;
        margin-top: 30px;
        max-width: 200px;
    
        .meter {
            max-height: 200px;
        }
        
        .meter-label {
            position: absolute;
            top: 65px;
            left: 39px;
            line-height: 0px;
            text-align: center;
            p {
                color: #fff;
                font-size: 30px;
            }
            span {
                color: white;
                font-size: 10px;
            }
        }
    }
    .chart-container {
        margin-top: 25px;
    }
    .table-container {
        color: white;
        margin-top: 50px;
        font-weight: 300;

        table {
            width: 100%;
            border-color: #fff !important;

            .table-head {
                margin-bottom: 5px;
                border-bottom: 1px solid #fff;
                font-weight: 400;
                :nth-child(2) {
                    text-align: center;
                }
                :nth-child(3) {
                    text-align: right;
                }
                }
                .table-row {
                    font-weight: 100;
                    :nth-child(2) {
                        text-align: center;
                    }
                    :nth-child(3) {
                        text-align: right;
                    }

            }
        }
    }
}
