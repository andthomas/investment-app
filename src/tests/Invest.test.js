import React from 'react';
import { shallow, configure } from 'enzyme';
import { Invest } from '../components/Invest';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Invest /> component test', () => {
    let wrapper;
    beforeEach(() => {
        const shares =
            [
                {
                    "id": "AAPL",
                    "name": "Apple Inc.",
                    "icon": "apple-logo_amrzzo.png"
                },
                {
                    "id": "BIDU",
                    "name": "Baidu Inc",
                    "icon": "baidu.png"
                },
            ];
        const selectedShare =
            [
                {
                    "id": "AAPL",
                    "name": "Apple Inc.",
                    "icon": "apple-logo_amrzzo.png"
                },
            ];
        wrapper = shallow(
            <Invest shareList={shares} selectedShares={selectedShare} />
        );
    })

    describe('<Invest /> rendering', () => {
        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
        it('Renders a company button with text and logo without crashing', () => {
            const aapl = wrapper.findWhere(node => node.key() === 'AAPL')
            expect(aapl.find('.share-name').text('Apple Inc.'));
            expect(aapl.find('.share-id').text('AAPL'));
            expect(aapl.find('.company-logo').prop('src')).toEqual('https://res.cloudinary.com/djq5ic5br/image/upload/c_scale,h_120/apple-logo_amrzzo.png');
        });
    });
    
    describe('<Invest /> interactions', () => {
        it('Filters the company list correctly', () => {
            const query = 'Baidu';
            wrapper.find('#searchbar').simulate('change', {target: {value: query}});
            expect(wrapper.state('visibleShares')).toHaveLength(1);
            expect(wrapper.state('visibleShares')[0].id).toEqual('BIDU');
        })
    })
});