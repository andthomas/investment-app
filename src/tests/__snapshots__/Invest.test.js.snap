// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Invest /> component test <Invest /> rendering renders correctly 1`] = `ShallowWrapper {}`;
